cgen generates the same struct twice when importing a distinct object

### Specification

When using `importc` and `nodecl` for an object that is just a distinct of another object, the compiler erroneously writes the same struct twice. When I edit the `.c` file to remove the duplicate, it works properly.

### Example

Compile with `--passL:-latomic`

```nim
type int128* = object
        hi*{.align:16.}, lo*: uint64
type cint128 {.importc: "__int128",nodecl.} = distinct int128

let x = int128(hi: 123'u64, lo: 321'u64)
let y = cast[cint128](x)
echo cast[int128](y)
```

### Actual Output

<!--
* any relevant output, crash, error messages, etc...
* we move pretty quick, so please ensure you have the latest version of the compiler
* sometimes the output isn't clear or in fact absent, please indicate the relevant aspects
-->

```
/home/user/.cache/nimskull/dupli_d/@mdupli.nim.c:42:8: error: redefinition of 'struct _G6int128_M5dupli'
   42 | struct _G6int128_M5dupli {
      |        ^~~~~~~~~~~~~~~~~
/home/user/.cache/nimskull/dupli_d/@mdupli.nim.c:38:8: note: originally defined here
   38 | struct _G6int128_M5dupli {
      |        ^~~~~~~~~~~~~~~~~

execution of an external program 'gcc -c  -w -fmax-errors=3   -I/home/user/nimskull/lib -I/tmp -o /home/user/.cache/nimskull/dupli_d/@mdupli.nim.c.o /home/user/.cache/nimskull/dupli_d/@mdupli.nim.c' failed with exit code '1'
```

### Expected Output

```
(hi: 123, lo: 321)
```

--- a/compiler/mir/mirtypes.nim
+++ b/compiler/mir/mirtypes.nim
@@ -1085,11 +1085,13 @@ proc handleImported(env: var TypeEnv, t: PType): TypeId =
   if t.sym != nil and sfImportc in t.sym.flags:
     # an imported type. It's wrapped in a ``tkImported``, referencing the
     # underlying type
-    let base = typeSymToMir(env):
+    let base =
       if t.kind in Skip:
-        t.lastSon.skipIrrelevant()
+        env.add t.lastSon.skipIrrelevant()
       else:
-        t
+        # create a type symbol without going through the cache or ``.importc``
+        # handling
+        typeSymToMir(env, t)
 
     discard getSize(env.config, t) # compute the sizes, alignments, and offsets
 

--- a/tests/ccgbugs/timport_distinct.nim
+++ b/tests/ccgbugs/timport_distinct.nim
@@ -1,0 +1,15 @@
+discard """
+  description: '''
+    Regression test for importing ``distinct`` types resulting in duplicate C
+    type definitions
+  '''
+"""
+
+type
+  Object = object
+  Imported {.importc: "int", nodecl.} = distinct Object
+
+var x: Object
+var y: Imported
+
+# two type definitions for `Object` were emitted

